internal/pkg/cli/deploy/workload.go

```diff
@@ -339,19 +339,27 @@ func (img ContainerImageIdentifier) customOrGitTag() string {
 	return img.GitShortCommitTag
 }
 
+// sidecarReferenceTag returns the tag that should be used to reference the image.
+func (img ContainerImageIdentifier) sidecarReferenceTag() string {
+	if img.GitShortCommitTag != "" {
+		return img.GitShortCommitTag
+	}
+	return img.uuidTag
+}
+
 func (d *workloadDeployer) uploadContainerImages(imgBuilderPusher imageBuilderPusher) (map[string]ContainerImageIdentifier, error) {
 	required, err := manifest.DockerfileBuildRequired(d.mft)
 	if err != nil {
 		return nil, err
 	}
-	if !required {
-		return nil, nil
-	}
 	// If it is built from local Dockerfile, build and push to the ECR repo.
 	buildArgsPerContainer, err := buildArgsPerContainer(d.name, d.workspacePath, d.image, d.mft)
 	if err != nil {
 		return nil, err
 	}
+	if !required && len(buildArgsPerContainer) == 0 {
```

Comment:
yeah you're right this definitely reads strange, I'd rather just have here:
```suggestion
	if len(buildArgsPerContainer) == 0 {
```
which makes things pretty obvious